### ISSUE TITLE
BUG: Render passes don't work correctly with subplots

### ISSUE THREAD

**Issue by larsoner** [pyvista/pyvista#4939]
I noticed a bug with `enable_ssao` and previously similar bugs with `enable_anti_aliasing`, which makes me think this is a general problem. This code:
```
import pyvista as pv

grid = pv.ImageData(dimensions=(5, 5, 5)).explode(0.2)
pl = pv.Plotter(shape=(1, 2))
pl.subplot(0, 0)
# pl.enable_ssao()  # 1. Okay
# pl.enable_anti_aliasing("ssaa")  # 2. Broken
pl.add_mesh(grid)
pl.subplot(0, 1)
# pl.enable_ssao(kernel_size=32)  # 3. Broken
# pl.enable_anti_aliasing("ssaa")  # 4. Broken
pl.add_mesh(grid)
pl.show()
```
Produces the following outputs, uncommenting no lines (0) or uncommenting only one of the other lines 1-4 numbered above:

| 0 (all commented) | 1 (ssao left col) | 2 (ssaa left col) | 3 (ssao right col) | 4 (ssaa right col) |
| - | - | - | - | - |
| ![image](https://github.com/pyvista/pyvista/assets/2365790/dd003dc9-b4de-4038-a33d-5489bcf49dfa) | ![image](https://github.com/pyvista/pyvista/assets/2365790/2dc8cb61-7ab7-4ea8-86ad-092067698b57) | ![image](https://github.com/pyvista/pyvista/assets/2365790/00d60b7c-ac04-4113-acf6-04d207b7eeaa) | ![image](https://github.com/pyvista/pyvista/assets/2365790/a54d6dba-254d-43cc-8bc3-04f2d1271410) | ![image](https://github.com/pyvista/pyvista/assets/2365790/7fe0a73d-8725-4c7a-88e0-aa8b86cd6ecd) |

Is this expected? If `enable_ssao` and some variants of `enable_anti_aliasing` are *not* supposed to work with multiple subplots then we should probably raise an error when people try to use them. If they are supposed to work then we should probably fix these errors. I assume this applies also to eye-dome lighting and maybe other methods. Ideally we could raise errors for all of these in multiplotter mode or fix (and image-test) all of them.

At the PyVistaQt end I guess we have MultiPlotter / https://github.com/pyvista/pyvistaqt/pull/78 for this, so maybe it's not supposed to be supported directly in `Plotter`?

**Comment by larsoner** [comment]
... at least in principle I don't see why it's not possible since [`vedo`'s SSAO example](https://github.com/marcomusy/vedo/blob/master/examples/basic/ssao.py) has one subplot with SSAO and the other without:

![Screenshot from 2023-09-20 13-41-57](https://github.com/pyvista/pyvista/assets/2365790/b3e9b4ab-4ba4-4003-8647-02f84899f733)

So maybe there are "just" some render window pass bugs to work out? Or maybe they use a different scheme for subplots somehow...?

**Comment by banesullivan** [comment]
FYI this may also be tracked in #256 and https://discourse.vtk.org/t/eye-dome-lighting-and-viewport/1249. My impression is that this is a VTK issue

**Comment by larsoner** [comment]
You're right, modifying the vedo example to use SSAO in the second plot breaks things (ignore the text, I only changed changed their example code to change which plot got SSAO enabled:

<img width="832" alt="Screenshot 2023-10-02 at 10 47 08 PM" src="https://github.com/pyvista/pyvista/assets/2365790/23fc60d9-3271-4e1e-b1e9-70e868a28b09">

Closing in favor of upstream vtk issue https://gitlab.kitware.com/vtk/vtk/-/issues/18849

**Comment by larsoner** [comment]
Reopening because this *might* be related to #256 and #8633 but is not fixed by #8633

**Comment by larsoner** [comment]
Opened https://gitlab.kitware.com/vtk/vtk/-/merge_requests/13411 upstream 🤞

**Comment by larsoner** [comment]
(Though maybe `pyvista` should `warn_external` if SSAO is attempted to be enabled on a multi-plot plotter, subject to a version-check for whatever VTK version the fix will land in?)

**Comment by larsoner** [comment]
Okay this [should work once VTK 9.7.0 is out](https://gitlab.kitware.com/vtk/vtk/-/merge_requests/13411). Should we emit an error or a warning on `VTK<9.7` if this is called and there is more than one subplot?

**Comment by user27182** [comment]
I’ve included this warning/fix as part of https://github.com/pyvista/pyvista/issues/8810

**Comment by banesullivan** [comment]
This is now solved by https://github.com/codimensional/pyvista-render-passes
